src/main/java/net/ripe/ipresource/Asn.java

```diff
@@ -117,13 +117,7 @@ protected int doHashCode() {
     protected int doCompareTo(IpResource obj) {
         if (obj instanceof Asn) {
             long otherValue = ((Asn) obj).longValue();
-            if (longValue() < otherValue) {
-                return -1;
-            } else if (longValue() > otherValue) {
-                return +1;
-            } else {
-                return 0;
-            }
+            return Long.compare(longValue(), otherValue);
```

Comment:
The call to `Long.compare(...)` requires Java 7. Just let me know if you need to support an older version. In that case, I will simply introduce a local variable to prevent the double call to `longValue()`.

Reply:
It is used in a bunch of projects in RIPE NCC by different teams, so it is hard to say if anyone still needs Java 6, but since this library is versioned and released, it's fine.

Reply:
Since the compiler source and target versions of this library are 1.7, let's assume this is not an issue :)